File: GAS/src/fi/quanfoxes/Keyword.java
package fi.quanfoxes;

import java.util.Objects;

public class Keyword {
    private KeywordType type;
    private String identifier;

    public Keyword(String identifier) {
        this(KeywordType.NORMAL, identifier);
    }

    public Keyword(KeywordType type, String identifier) {
        this.type = type;
        this.identifier = identifier;
    }

    public KeywordType getType() {
        return type;
    }

    public String getIdentifier() {
        return identifier;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Keyword)) return false;
        Keyword keyword = (Keyword) o;
        return Objects.equals(type, keyword.type) &&
               Objects.equals(identifier, keyword.identifier);
    }

    @Override
    public int hashCode() {
        return Objects.hash(identifier, type);
    }
}

File: Vivid/Lexer/Keyword.cs
public enum KeywordType
{
	MODIFIER,
	FLOW,
	NORMAL
}

public class Keyword
{
	public KeywordType Type { get; }
	public string Identifier { get; }

	public Keyword(string identifier, KeywordType type = KeywordType.NORMAL) : this(type, identifier) { }

	public Keyword(KeywordType type, string identifier)
	{
		Type = type;
		Identifier = identifier;
	}

	public T To<T>() where T : Keyword
	{
		return (T)this;
	}
}